pkg/processor/procjs/processor.go

```diff
@@ -28,6 +28,20 @@ const (
 	entrypoint = "process"
 )
 
+// jsRecord is an intermediary representation of record.Record that is passed to
+// the JavaScript transform. We use this because using record.Record would not
+// allow us to modify or access certain data (e.g. metadata or structured data).
+type jsRecord struct {
```

Comment:
This part needs special attention. An intermediary record struct was introduced to fix handling of `Record.Metadata` in the JS processor, that field had essentially the same problem as structured data (see https://github.com/ConduitIO/conduit/issues/468). That issue should also be fixed now, since structured data is transformed into a simple `map[string]interface{}` when it's passed to the JS function.